debug.js

```diff
@@ -55,74 +67,145 @@ function selectColor() {
+      enabled.diff = ms;
+      enabled.prev = prevTime;
+      enabled.curr = curr;
+      prevTime = curr;
+
+      // add the `color` if not set
+      if (null == enabled.useColors) enabled.useColors = exports.useColors();
+      if (null == enabled.color && enabled.useColors) enabled.color = selectColor();
 
-  // define the `enabled` version
-  function enabled() {
+      var args = Array.prototype.slice.call(arguments);
 
-    var self = enabled;
+      args[0] = exports.coerce(args[0]);
 
-    // set `diff` timestamp
-    var curr = +new Date();
-    var ms = curr - (prevTime || curr);
-    self.diff = ms;
-    self.prev = prevTime;
-    self.curr = curr;
-    prevTime = curr;
+      if ('string' !== typeof args[0]) {
+        // anything else let's inspect with %o
+        args = ['%o'].concat(args);
+      }
+
+      // apply any `formatters` transformations
+      var index = 0;
+      args[0] = args[0].replace(/%([a-z%])/g, function(match, format) {
+        // if we encounter an escaped % then don't increase the array index
+        if (match === '%%') return match;
+        index++;
+        var formatter = exports.formatters[format];
+        if ('function' === typeof formatter) {
+          var val = args[index];
+          match = formatter.call(enabled, val);
+
+          // now we need to remove `args[index]` since it's inlined in the `format`
+          args.splice(index, 1);
+          index--;
+        }
+        return match;
+      });
+
+      if ('function' === typeof exports.formatArgs) {
+        args = exports.formatArgs.apply(enabled, args);
+      }
+      var logFn = enabled.log || exports.log || console.log.bind(console);
+      logFn.apply(enabled, args);
+    }
+    enabled.enabled = true;
 
-    // add the `color` if not set
-    if (null == self.useColors) self.useColors = exports.useColors();
-    if (null == self.color && self.useColors) self.color = selectColor();
+    var fn = exports.enabled(namespace) ? enabled : disabled;
 
-    var args = Array.prototype.slice.call(arguments);
+    fn.namespace = namespace;
 
-    args[0] = exports.coerce(args[0]);
+    // Fake release() method.
+    fn.release = function() {};
 
-    if ('string' !== typeof args[0]) {
-      // anything else let's inspect with %o
-      args = ['%o'].concat(args);
-    }
+    return fn;
+  }
 
-    // apply any `formatters` transformations
-    var index = 0;
-    args[0] = args[0].replace(/%([a-z%])/g, function(match, format) {
-      // if we encounter an escaped % then don't increase the array index
-      if (match === '%%') return match;
-      index++;
-      var formatter = exports.formatters[format];
-      if ('function' === typeof formatter) {
-        var val = args[index];
-        match = formatter.call(self, val);
-
-        // now we need to remove `args[index]` since it's inlined in the `format`
-        args.splice(index, 1);
-        index--;
+  // Return a dynamic debug instance.
+  else {
+    function dynamic() {
```

Comment:
Right. We can talk about code optimization if the PR is at least considered by the author, which sai nothing about it.